runtime/src/main/java/org/corfudb/runtime/collections/CorfuStore.java

```diff
@@ -216,6 +216,17 @@ public TxnContext txn(@Nonnull final String namespace, IsolationLevel isolationL
                 false);
     }
 
+    @Nonnull
+    public ScopedTransaction scopedTxn(
+            @Nonnull final String namespace, IsolationLevel isolationLevel,
+            Table<?, ?, ?>... tables) {
+        return new ScopedTransaction(
```

Comment:
We need to surround this block within an IRetry context to handle the case where a `TrimmedException` is thrown due to the snapshot's version not being in the cache.
In our scope transaction implementation we did it as follows...
`try {
            return IRetry.build(IntervalRetry.class, () -> {
                try {
                    return new ScopedTransaction(
                            this.runtime,
                            namespace,
                            isolationLevel,
                            tables);
                } catch (TrimmedException e) {
                    log.error("Error while attempting to snapshot tables {}.", tables, e);
                    throw new RetryNeededException();
                }
            }).run();
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }`
Perhaps something similar to avoid the `TrimmedException`?